app/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryMongoActor.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.actors
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{Actor, Props, Scheduler}
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryManager.RetryTaxEnrolmentRequest
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryMongoActor.{DeleteTaxEnrolmentRetryRequest, GetAllNonFailedTaxEnrolmentRetryRequests, UpdateTaxEnrolmentRetryStatusToFail}
+import uk.gov.hmrc.cgtpropertydisposals.models.TaxEnrolmentRequest
+import uk.gov.hmrc.cgtpropertydisposals.repositories.TaxEnrolmentRetryRepository
+import uk.gov.hmrc.cgtpropertydisposals.util.Logging
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.duration.FiniteDuration
+import scala.language.postfixOps
+
+class TaxEnrolmentRetryMongoActor(
+  scheduler: Scheduler,
+  taxEnrolmentRetryRepository: TaxEnrolmentRetryRepository,
+  maxStaggerTime: Int)
+    extends Actor
+    with Logging {
+
+  import context.dispatcher
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  override def receive: Receive = {
+
+    case GetAllNonFailedTaxEnrolmentRetryRequests =>
+      val sender_ = sender()
+      taxEnrolmentRetryRepository.getAllNonFailedEnrolmentRequests().value.map {
+        case Left(error) => logger.warn(s"Error recovering tax enrolment retry requests: [$error]")
```

Comment:
if you import `import uk.gov.hmrc.cgtpropertydisposals.util.Logging._` you can pass the error as a second argument to `warn` and that will log the stack trace as well